expression/column.go

```diff
@@ -273,7 +273,7 @@ func (col *Column) VecEvalReal(ctx sessionctx.Context, input *chunk.Chunk, resul
 
 // VecEvalString evaluates this expression in a vectorized manner.
 func (col *Column) VecEvalString(ctx sessionctx.Context, input *chunk.Chunk, result *chunk.Column) error {
-	if col.RetType.Hybrid() {
+	if col.RetType.Hybrid() || ctx.GetSessionVars().StmtCtx.PadCharToFullLength {
```

Comment:
Why change this line? Any test case cover this.

Reply:
How about using another PR to fix this problem?

Reply:
in PAD_CHAR_TO_FULL_LENGTH sql_mod, we need to pad the string to specified length, but we don't do it in ` func VecEvalString`, this will result in the fail of below test case.

```
const_test.go:272:
    r.Check(testkit.Rows("0 10"))
/deps/go/src/github.com/pingcap/tidb/util/testkit/testkit.go:62:
    res.c.Assert(resBuff.String(), check.Equals, needBuff.String(), res.comment)
... obtained string = "[1 10]\n"
... expected string = "[0 10]\n"
... sql:SELECT a='xy        ', char_length(a) FROM t1;, args:[]
```

I will fix it in a new PR.